### ISSUE TITLE
Feature of checking for pattern matching on case changing enums (discussion).

### ISSUE THREAD

**Issue by agentgt** [typetools/checker-framework#8033]
One of the challenges of pattern matching enums is that not all enums are designed for exhaustive pattern matching.

I have written my own annotation that you put on an enum to indicate that you should not rely on exhaustive pattern matching:
 https://jstach.io/doc/rainbowgum/current/apidocs/io.jstach.rainbowgum.annotation/io/jstach/rainbowgum/annotation/CaseChanging.html

The idea is to avoid the issue where version one we have

```java
enum MyEnum {
 A,
 B
}
```

Consumer of library writes

```java
MyEnum e;
switch(e) {
  case A -> ...;
  case B -> ...;
}
```

And then version 2 a mostly backward compatible version (lets ignore the debate what is a major, minor, etc semver policy and binary vs source compat):

```java
enum MyEnum {
 A,
 B,
 C
}
```

Then the client will get an unexpected failure for `C`.

Any way I was going to take a stab on implementing this checker on my own to try to improve my knowledge of the checkerframework and thought I would mention it.

**Comment by smillst** [comment]
This sounds like an interesting idea!  Make sure you read https://checkerframework.org/manual/#creating-tips.
